kore/src/main/scala/org/kframework/attributes/Att.scala

```diff
@@ -4,36 +4,40 @@ import java.util.Optional
 
-  def contains(cls: Class[_]): Boolean = att.contains((cls.getName, cls))
-  def contains(key: String): Boolean = att.contains((key, classOf[String]))
-  def contains(key: String, cls: Class[_]): Boolean = att.contains((key, cls))
+  def contains(cls: Class[_]): Boolean = att.contains((cls.getName, cls.getName))
+  def contains(key: String): Boolean = att.contains((key, Att.stringClassName))
+  def contains(key: String, cls: Class[_]): Boolean = att.contains((key, cls.getName))
 
   def get[T](key: Class[T]): T = getOption(key).get
   def get(key: String): String = getOption(key).get
   def get[T](key: String, cls: Class[T]): T = getOption(key, cls).get
-  def getOption(key: String): Option[String] = att.get((key, classOf[String])).asInstanceOf[Option[String]]
-  def getOption[T](key: Class[T]): Option[T] = att.get((key.getName, key)).asInstanceOf[Option[T]]
-  def getOption[T](key: String, cls: Class[T]): Option[T] = att.get((key, cls)).asInstanceOf[Option[T]]
+  def getOption(key: String): Option[String] = att.get((key, Att.stringClassName)).asInstanceOf[Option[String]]
+  def getOption[T](key: Class[T]): Option[T] = att.get((key.getName, key.getName)).asInstanceOf[Option[T]]
+  def getOption[T](key: String, cls: Class[T]): Option[T] = att.get((key, cls.getName)).asInstanceOf[Option[T]]
   def getOptional(key: String): Optional[String] = optionToOptional(getOption(key))
   def getOptional[T](key: Class[T]): Optional[T] = optionToOptional(getOption(key))
   def getOptional[T](key: String, cls: Class[T]): Optional[T] = optionToOptional(getOption(key, cls))
 
-  private def optionToOptional[T](option: Option[T]): Optional[T] = option match { case None => Optional.empty(); case Some(x) => Optional.of(x); }
+  private def optionToOptional[T](option: Option[T]): Optional[T] =
+    option match {case None => Optional.empty(); case Some(x) => Optional.of(x);}
 
   def add(key: String): Att = add(key, "")
-  def add(key: String, value: String): Att = add(key, classOf[String], value)
-  def add[T](key: String, cls: Class[T], value: T): Att = Att(att + ((key, cls) -> value))
-  def add[T](key: Class[T], value: T): Att = add(key.getName, key, value)
+  def add(key: String, value: String): Att = add(key, Att.stringClassName, value)
+  def add[T](key: Class[T], value: T): Att = add(key.getName, key.getName, value)
+  def add[T](key: String, cls: Class[T], value: T): Att = add(key, cls.getName, value)
+  private def add[T](key: String, clsStr: String, value: T): Att = Att(att + ((key, clsStr) -> value))
   def addAll(thatAtt: Att) = Att(att ++ thatAtt.att)
-  def addAll(thatAtt: java.util.Map[String, String]): Att = Att(immutable(thatAtt).map { case (k, v) => ((k, classOf[String]), v)}.toMap)
 
-  def remove(key: String): Att = remove(key, classOf[String])
-  def remove(key: Class[_]): Att = remove(key.getName, key)
-  def remove(key: String, cls: Class[_]): Att = Att(att - ((key, cls)))
```

Comment:
What happened to this overload of remove?

Reply:
I don't understand the question. There are still 3 version of `remove`.

Reply:
There is no longer a String, Class overload.

Reply:
I inlined that one - it was only used inside `Att`.

Reply:
It can be put back later if needed. I'll add it back of course.